test/uniport-tests.js

```diff
@@ -0,0 +1,502 @@
+/* -------------------------------- REQUIRES -------------------------------- */
+
+var async       = require("async");
+var assert      = require('assert');
+var assert      = require('assert');
+var UInt160     = require("ripple-lib").UInt160;
+var Remote      = require('ripple-lib').Remote;
+var Server      = require('ripple-lib').Server;
+var Request     = require('ripple-lib').Request;
+var testutils   = require("./testutils");
+var config      = testutils.init_config();
+var http        = require('http');
+var request     = require('request');
+
+/* --------------------------------- CONFIG --------------------------------- */
+
+// So we can connect to websocket ports
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+
+var configurations = {
+  // All the tests run on one invokation of rippled. It's possible for you to:
+  use_own_rippled : false,
+
+  // We have one main websocket connection, that we inherit from
+  // `testutils.build_setup`. We use that, configured on a non admin, non ssl
+  // port, to determine readiness for tests on the other ports. This Remote's
+  // trace is configured in the normal way, however we can set trace for each of
+  // the test connections on the various ports.
+  remote_trace : false,
+
+  // There's a bug where we can't subscribe to the admin ports requiring
+  // credentials, even though subscribe has Role::USER. `Remote` always issues a
+  // subscribe command after connect(). We can just add the credentials to let
+  // the tests go further to test that admin requests without credentials are
+  // not honoured.
+  add_credentials_to_ws_subscribe: false,
+
+  // WebSocket connections honour admin=no regardless of ip.
+  // Yet http with accepts admin commands from any whitelisted ip.
+  // 127.0.0.1 is hardcoded in cpp code and added to [rpc_admin_allow].
+  expect_http_to_honour_admin_no: process.env.HTTP_HONOUR_ADMIN_NO || true,
+
+  // We can be a bit more exhaustive, but it will cost us 2 seconds, unless
+  // we use `skip_tests_matching` to just `test.skip` them.
+  define_redundant_tests : process.env.REDUNDANT_UNIPORT_TESTS || true,
+
+  skip_tests_matching : [
+    // /redundant/
+
+    // /can not issue/,
+    // /http/
+  ],
+
+  skip_tests_not_matching : [
+    // /wrong password/,
+    // /ws/
+  ],
+
+  skip_ports_not_matching : [
+    // /admin/,
+    // /password/,
+    // /http/
+  ],
+
+  skip_ports_matching : [
+    // /admin/,
+    // /password/,
+    // /http/
+  ]
+}
+
+/* --------------------------------- HELPERS -------------------------------- */
+
+var eachItem = function (o, f) {
```

Comment:
Why not just use the other form of function?  `function eachItem(o, f) {` - this is more idiomatic than a `var` statement.